org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/ContentModelFormatterParticipant.java

```diff
@@ -40,8 +43,11 @@ public class ContentModelFormatterParticipant implements IFormatterParticipant {
 
 	private final ContentModelManager contentModelManager;
 
+	private final Map<String, Collection<CMDocument>> cmDocumentsCache;
+
 	public ContentModelFormatterParticipant(ContentModelManager contentModelManager) {
 		this.contentModelManager = contentModelManager;
+		cmDocumentsCache = Collections.synchronizedMap(new HashMap<>());

```

Comment:
The cache should be not hosted in ContentModelFormatterParticpant but it should be hosted in a Map initialized by the formatter. Because when you implement a cache, there is the problem of the cache evict.


I will change the signature of getFormatElementCategory like this:

```java
public FormatElementCategory getFormatElementCategory(DOMElement element,
			XMLFormattingConstraints parentConstraints, Map<String, Object> formattingContext, SharedSettings sharedSettings) {
```

The formattingContext is a kind of cache that any formatter participant could use. After that I will store your map which stores mapping between namespaceURI and CMDocument collections not directly in formattingContext, The formattingContext should store with a given key (ex : CM_DOCUMENTS_MAPPINGS_KEY ) your cmDocumentsCache.